Rely on official types

yringler opened the pull request:
Fixes #44

yringler commented:
This is a huge PR in terms of effected changes, but not in terms of complexity.
I think main review should be in builds and tests passing.

AckerApple commented on dist/package.json:
```diff
@@ -1,6 +1,6 @@
 {
   "name": "stripe-angular",
-  "version": "1.4.2",
+  "version": "1.4.3",
```
Make this 1.5.0 please

AckerApple commented:
Looking pretty good. I am on PTO so no action from me until next week

yringler commented:
Updated version, and removed rest of custom types.
I ran into one issue: in [/app/src/components/app.component.ts](https://github.com/yringler/stripe-angular/blob/9dea81f6e4623b1d498d7eed9a05f59106289941/app/src/components/app.component.ts#L115), the official types are missing the metadata property, so I had to cast it... I know there's a way to extend interfaces etc, I gave it a shot, but I'm not so good at that... I opened a PR DefinitelyTyped/DefinitelyTyped/pull/50386 , will see what happens.
In the meantime I added a cast to this code

AckerApple commented:
Just a note I am actively working on this project today so any notes, thoughts, and changes are best expressed today.

I will review and merge today

yringler commented:
One note - I think this is built on an older version of angular? I know to get a part of this to build I had to specify the typescript version in package.json, because the current LTS was too new.
It would be nice if everything was updated.
